Makefile

```diff
@@ -96,39 +97,49 @@ analyse_canary: ## run php-cs-fixer and phpstan (dark-canary)
 
 ##@ DEPLOY
 
-ACCOUNT=$(shell aws sts get-caller-identity --profile=poser | jq -r '.Account')
-VER=$(shell date +%s)
-
-deploy_prod: ## deploy to prod
-	# docker_image_phpfpm
-	docker build \
-		-t $(ACCOUNT).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:phpfpm-$(VER) \
-		-f sys/docker/alpine-phpfpm/Dockerfile .
-	docker push $(ACCOUNT).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:phpfpm-$(VER)
-
-	# docker_image_phpfpm_darkcanary
+.docker_img_deps:
+	docker build -t jsbuilder -f sys/docker/common/Dockerfile.assets .
+	docker build -t awsbuilder -f sys/docker/common/Dockerfile.aws-ecs-ci .
+
+AWS_PROFILE ?= poser
+AWS_ACCOUNT_ID ?= $(shell aws sts get-caller-identity --profile=$(AWS_PROFILE) | jq -r '.Account')
+PREVIOUS_TAG=$(shell git ls-remote --tags 2>&1 | awk '{print $$2}' | sort -r | head -n 1 | cut -d "/" -f3)
+
+deploy_prod: .docker_img_deps ## deploy to prod
+	git fetch --all --tags > /dev/null; \
+	git log --pretty=format:"- %B" ${START_LOG}..HEAD | tr '\r' '\n' | grep -Ev '^$$' > CHANGELOG; \
+	sed -i 's/^*/-/' CHANGELOG; sed -i 's/^[ \t]*//' CHANGELOG; sed -i 's/^-[ \t]*//' CHANGELOG; sed -i 's/^-[ \t]*//' CHANGELOG; sed -i 's/^/ - /' CHANGELOG; \
+	aws ecr get-login-password --profile $$AWS_PROFILE | docker login --password-stdin -u AWS $(AWS_ACCOUNT_ID).dkr.ecr.eu-west-1.amazonaws.com; \
+	VER=$(shell date +%s); \
+    docker build \
+		-t $(AWS_ACCOUNT_ID).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:phpfpm-$$VER \
+		-f sys/docker/alpine-phpfpm/Dockerfile .; \
+	docker push $(AWS_ACCOUNT_ID).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:phpfpm-$$VER; \
 	docker build \
-		-t $(ACCOUNT).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:phpfpm8-$(VER) \
-		-f sys/docker/alpine-phpfpm8/Dockerfile .
-	docker push $(ACCOUNT).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:phpfpm8-$(VER)
-
-	# docker_image_nginx
+		-t $(AWS_ACCOUNT_ID).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:phpfpm8-$$VER \
+		-f sys/docker/alpine-phpfpm8/Dockerfile .; \
+	docker push $(AWS_ACCOUNT_ID).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:phpfpm8-$$VER; \
 	docker build \
-		-t $(ACCOUNT).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:nginx-$(VER) \
-		-f sys/docker/alpine-nginx/Dockerfile .
-	docker push $(ACCOUNT).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:nginx-$(VER)
-
+		-t $(AWS_ACCOUNT_ID).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:nginx-$$VER \
+		-f sys/docker/alpine-nginx/Dockerfile .; \
+	docker push $(AWS_ACCOUNT_ID).dkr.ecr.eu-west-1.amazonaws.com/badge-poser:nginx-$$VER; \
 	cat sys/cloudformation/parameters.prod.json \
-		| jq '.[18].ParameterValue="$(VER)" | .[19].ParameterValue="$(VER)"' \
-		| tee sys/cloudformation/parameters.prod.json
-
+		| sed 's/{"ParameterKey": "EcrImageTagNginx", "ParameterValue": ".+"}/{"ParameterKey": "EcrImageTagNginx", "ParameterValue": "'$$VER'"}/'
```

Comment:
sed allows multiple operations in one invoke

Reply:
fixed